### PR TITLE
Stop the serving info setter replacing the cert from the cert updater.

### PR DESCRIPTION
**Pull request by howbazaar**
There was a race between the serving info setter and the cert updater.

Most of the time the serving info would run first, and set the cert based on the copy in the database. But sometimes the cert updater would run first and the new cert would be overwritten by the cert stored in the database.

## QA

Was able to reproduce the failure by running the test repeatedly while the machine was very busy.
Can no longer reproduce the failure.

### CODE REVIEW

**Comment by howbazaar** [comment]
Fixes https://bugs.launchpad.net/juju/+bug/1466514

File: cmd/jujud/agent/machine/servinginfo_setter.go [hunk 1]
```diff
@@ -57,12 +57,21 @@ func ServingInfoSetterManifold(config ServingInfoSetterConfig) dependency.Manifo
...
 			for _, job := range machine.Jobs() {
 				if job.NeedsState() {
 					info, err := apiState.StateServingInfo()
 					if err != nil {
 						return nil, errors.Errorf("cannot get state serving info: %v", err)
 					}
+					if hasInfo {
+						// Use the existing Cert as it may have been updated already
+						// by the cert updater worker to have this machine's IP address
+						// as part of the info. This changed cert is never put back into
```
Comment (mjs) [inline cmd/jujud/agent/machine/servinginfo_setter.go]: should that be "as part of the *cert*"?

Comment (howbazaar) [inline cmd/jujud/agent/machine/servinginfo_setter.go]: Probably.

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**